sql/parser/builtins.go

```diff
@@ -761,3 +774,12 @@ func datumToRawString(datum Datum) (string, error) {
 
 	return "", fmt.Errorf("argument type unsupported: %s", datum.Type())
 }
+
+func round(x float64, n int64) (Datum, error) {
+	if n < 0 || n > 50 {
```

Comment:
Oh, you're right! Why do you think higher values than 308 should be allowed? `round(1.123e100, math.MaxInt32)` took 14 seconds in my test.

Should I set the upper limit to `MaxInt32`? Corresponding to the maximum number that is guaranteed to fit in an `int`?

Unfortunately, I merged the PR already, happy to apply a follow-up fix.